controllers/watcherapi_controller.go

```diff
@@ -444,10 +450,88 @@ func (r *WatcherAPIReconciler) ensureServiceExposed(
 	return apiEndpoints, ctrl.Result{}, nil
 }
 
+func (r *WatcherAPIReconciler) ensureKeystoneEndpoint(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *watcherv1beta1.WatcherAPI,
+	apiEndpoints map[string]string,
+) (ctrl.Result, error) {
+	Log := r.GetLogger(ctx)
+	Log.Info(fmt.Sprintf("Defining WatcherAPI KeystoneEndpoint '%s'", instance.Name))
+
+	endpointSpec := keystonev1.KeystoneEndpointSpec{
+		ServiceName: watcher.ServiceName,
+		Endpoints:   apiEndpoints,
+	}
+	endpoint := keystonev1.NewKeystoneEndpoint(
+		watcher.ServiceName,
+		instance.Namespace,
+		endpointSpec,
+		getAPIServiceLabels(),
+		r.RequeueTimeout,
+	)
+	ctrlResult, err := endpoint.CreateOrPatch(ctx, helper)
+	if err != nil {
+		return ctrlResult, err
+	}
+
+	c := endpoint.GetConditions().Mirror(condition.KeystoneEndpointReadyCondition)
+	if c != nil {
+		instance.Status.Conditions.Set(c)
+	}
+
+	if (ctrlResult != ctrl.Result{}) {
+		// We can ignore RequeueAfter as we are watching the KeystoneEndpoint
+		// resource
+		return ctrlResult, nil
+	}
+
+	return ctrl.Result{}, nil
```

Comment:
I think you can remove the `if` block here and just `return ctrlResult, nil`

Reply:
true, good catch, I'll push the change with the updated functional test